src/main/java/org/moqucu/games/nightstalker/sprite/enemy/GreyRobot.java

```diff
@@ -244,4 +247,65 @@ public Line getLineOfSight() {
                 furthestReachableNode.getY()
         );
     }
+
```

Comment:
I am myself undecided: below code could be moved into the game loop in the sense that it decides when one object is in the line of sight of another object. But then I am not sure either, if this is truly object-oriented? The general idea is to defer decisions so far to the bottom of the object tree that something that is truly closest to the desired action, will be responsible for making it. You could argue that the robot has a sensor that will be able to spot of things are in the line of sight and then make a decision. Maybe, the game loop puts things into the line of sight and then the individual actors decide...